HpToolsLauncher/Runners/FileSystemTestsRunner.cs

```diff
@@ -1,549 +1,580 @@
+                    }
+                    catch (Exception ex)
+                    {
+                        runResult = new TestRunResults
+                        {
+                            TestState = TestState.Error,
+                            ErrorDesc = ex.Message,
+                            TestName = test.TestName,
+                            TestPath = test.TestPath
+                        };
+                    }
+
+                    //get the original source for this test, for grouping tests under test classes
+                    runResult.TestGroup = test.TestGroup;
+
+                    activeRunDesc.TestRuns.Add(runResult);
+
+                    //if fail was terminated before this step, continue
+                    if (runResult.TestState != TestState.Failed)
+                    {
+                        if (runResult.TestState != TestState.Error)
+                        {
+                            Helper.GetTestStateFromReport(runResult);
+                        }
+                        else
+                        {
+                            if (string.IsNullOrEmpty(runResult.ErrorDesc))
+                            {
+                                runResult.ErrorDesc = RunCancelled() ? Resources.ExceptionUserCancelled : Resources.ExceptionExternalProcess;
+                            }
+                            runResult.ReportLocation = null;
+                            runResult.TestState = TestState.Error;
+                        }
+                    }
+
+                    if (runResult.TestState == TestState.Passed && runResult.HasWarnings)
+                    {
+                        runResult.TestState = TestState.Warning;
+                        ConsoleWriter.WriteLine(Resources.FsRunnerTestDoneWarnings);
+                    }
+                    else
+                    {
+                        ConsoleWriter.WriteLine(string.Format(Resources.FsRunnerTestDone, runResult.TestState));
+                        if (runResult.TestState == TestState.Error)
+                        {
+                            ConsoleWriter.WriteErrLine(runResult.ErrorDesc);
+                        }
+                    }
+
+                    UpdateCounters(runResult.TestState);
+                    var testTotalTime = (DateTime.Now - testStart).TotalSeconds;
+                    ConsoleWriter.WriteLine(DateTime.Now.ToString(Launcher.DateFormat) + " Test completed in " + ((int)Math.Ceiling(testTotalTime)).ToString() + " seconds: " + runResult.TestPath);
+                    if (!string.IsNullOrWhiteSpace(runResult.ReportLocation))
+                    {
+                        ConsoleWriter.WriteLine(DateTime.Now.ToString(Launcher.DateFormat) + " Test report is generated at: " + runResult.ReportLocation + NEW_LINE_AND_DASH_SEPARATOR);
+                    }
+
+                    // break to cancel the remaining tests

```

Comment:
now since you're using one `foreach` loop, and no break logic anymore, here the comment should be: 
```cs
// skip remaining tests if the current test is failure or error and cancelRunOnFailure is true
```